framework/wazuh/core/task_dispatcher.py

```diff
@@ -0,0 +1,365 @@
+        except json.decoder.JSONDecodeError:
+            e = exception.WazuhInternalError(3036)
+            if self.debug:
+                raise
+            self.logger.error(f'{e.message}')
+            return e
+        except exception.WazuhInternalError as e:
+            if self.debug:
+                raise
+            self.logger.error(f'{e.message}', exc_info=not isinstance(e, exception.WazuhClusterError))
+            return e
+        except exception.WazuhError as e:
+            return e
+        except Exception as e:
+            if self.debug:
+                raise
+
+            self.logger.error(f'Unhandled exception: {str(e)}', exc_info=True)
+            return exception.WazuhInternalError(1000)
+
+    @staticmethod
+    def run_local(f, f_kwargs, rbac_permissions, current_user, origin_module, rbac_manager):
+        """Run framework SDK function locally in another process."""
+        common.rbac.set(rbac_permissions)
+        common.current_user.set(current_user)
+        common.origin_module.set(origin_module)
+        common.rbac_manager.set(rbac_manager)
+        data = f(**f_kwargs)
+        common.reset_context_cache()
+        return data
+
+    async def execute_local_request(self) -> str:  # noqa: C901
+        """Execute an API request locally.
+
+        Returns
+        -------
+        str
+            JSON response.
+        """
+        try:
+            if self.f_kwargs.get('agent_list') == '*':
+                del self.f_kwargs['agent_list']
+
+            before = time.time()
+
+            timeout = self.api_request_timeout if not self.wait_for_complete else None
+
+            try:
+                if self.is_async:
+                    task = self.run_local(
+                        self.f,
+                        self.f_kwargs,
+                        self.rbac_permissions,
+                        self.current_user,
+                        self.origin_module,
+                        self.rbac_manager,
+                    )
+
+                else:
+                    loop = asyncio.get_event_loop()
+                    if 'thread_pool' in pools:
+                        pool = pools.get('thread_pool')
+                    elif self.f.__name__ in authentication_funcs:
+                        pool = pools.get('authentication_pool')
+                    else:
+                        pool = pools.get('process_pool')
+
+                    task = loop.run_in_executor(
+                        pool,
+                        partial(
+                            self.run_local,
+                            self.f,
+                            self.f_kwargs,
+                            self.rbac_permissions,
+                            self.current_user,
+                            self.origin_module,
+                            self.rbac_manager,
+                        ),
+                    )
+                try:
+                    self.debug_log('Starting to execute request locally')
```

Comment:
Let's take advantage of the refactor to give a little more context on what is being executed

```suggestion
                    self.debug_log(f'Starting to execute request `{self.f.__name__}` locally')
```

Same with the logs below. Perhaps we could add an ID to match the logs corresponding to the same request.

Reply:
Fixed at ec34e60bc08af9548064954e3a9db834fd905907